src/components/vm/nics/vmNicsCard.tsx

```diff
@@ -269,6 +263,111 @@ interface IpInterface {
     source: string,
 }
 
+function sort_ips_by_source(ips: IpInterface[]): IpInterface[] {
+    // Setup a preferred order of IP/SUBNET-MASK we want to add.
+    // Helps ensure that the IP subnet mask are taken from the right source.
+    // agent > lease > arp
+    const source_order: {[index: string]: number} = {
+        agent: 1,
+        lease: 2,
+        arp: 3
+    };
+    return ips.sort((a, b) => source_order[a.source] - source_order[b.source]);
+}
+
+const IPDescriptions = ({
+    ipInterfaces,
+    id,
+    networkId,
+} : {
+    ipInterfaces: IpInterface[],
+    id: string,
+    networkId: number,
+}) => {
+    const inetIps: IpAddress[] = [];
+    const inet6Ips: IpAddress[] = [];
+
+    ipInterfaces.forEach(ipInterface => {
+        ipInterface.ips.forEach(inet => {
+            // Separate out IPv4 and IPv6 and don't add duplicate IPs.
+            if (inet.type === IpAddressVersion.v4 && !inetIps.some(inetIp => inetIp.ip == inet.ip)) {
+                inetIps.push(inet);
+            } else if (inet.type === IpAddressVersion.v6 && !inet6Ips.some(inet6Ip => inet6Ip.ip == inet.ip)) {
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2324-9e6bb0ec-20250918-065447-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)